metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala

```diff
@@ -0,0 +1,227 @@
+
+  private def readStacktraceFile(path: AbsolutePath): List[String] = {
+    Source.fromFile(path.toString).getLines().toList
+  }
+
+  def definition(
+      path: AbsolutePath,
+      position: l.Position
+  ): java.util.List[Location] = {
+    val lineOpt = readStacktraceFileLine(path, position.getLine)
+    lineOpt match {
+      case Some(line) =>
+        getSymbolLocationFromLine(line).toList.asJava
+      case None =>
+        java.util.Collections.emptyList()
+    }
+  }
+
+  private def adjustPosition(lineNumber: Int, pos: l.Position): Unit = {
+    pos.setLine(lineNumber)
+    pos.setCharacter(0)
+  }
+
+  private def tryGetLineNumberFromStacktrace(line: String): Try[Int] = {
+    Try(
+      // number is 1-based, but editors are 0-based
+      Integer.valueOf(
+        line.substring(line.indexOf(":") + 1, line.indexOf(")"))
+      ) - 1
+    )
+  }
+
+  private def convert(symbol: String): String = {
+    val components =
+      symbol.split('.').map(_.replace("$", "").replace("<init>", "init")).toList
+    val scope = components.dropRight(1)
+    val method = components.last
+    scope.mkString("/") ++ "." ++ method ++ "()."
+  }
+
+  def stacktraceLenses(content: List[String]): Seq[l.CodeLens] = {
+    (for {
+      (line, row) <- content.zipWithIndex
+      if line.trim.startsWith("at ")
+      location <- getSymbolLocationFromLine(line)
+      range = new l.Range(new l.Position(row, 0), new l.Position(row, 0))
+    } yield makeGotoLocationCodeLens(location, range)).toSeq
+  }
+
+  private def makeGotoLocationCodeLens(
+      location: l.Location,
+      range: l.Range
+  ): l.CodeLens = {
+    new l.CodeLens(
+      range,
+      new l.Command(
+        s"$gotoicon open",
+        ServerCommands.GotoLocationForPosition.id,
+        List[Object](location: Object, java.lang.Boolean.TRUE).asJava
+      ),
+      null
+    )
+  }
+
+  private def analyzeStackTrace(
+      stacktrace: String
+  ): Option[l.ExecuteCommandParams] = {
+    if (commandsInHtmlSupported) {
+      Some(makeHtmlCommandParams(stacktrace))
+    } else {
+      val path = workspace.resolve(Directories.stacktrace)
+      val pathStr = path.toFile.toString
+
+      path.toFile.createNewFile()
+      val fw = new FileWriter(pathStr)
+      try {
+        fw.write(stacktrace)
+      } finally {
+        fw.close()
+      }
+      val pos = new l.Position(0, 0)
+      val range = new l.Range(pos, pos)
+      val location = new l.Location(pathStr, range)
+      Some(makeGotoCommandParams(location))
+    }
+  }
+
+  private def makeGotoCommandParams(
+      location: Location
+  ): l.ExecuteCommandParams = {
+    new l.ExecuteCommandParams(
+      ClientCommands.GotoLocation.id,
+      List[Object](location, java.lang.Boolean.TRUE).asJava
+    )
+  }
+
+  private def getSymbolLocationFromLine(line: String): Option[l.Location] = {
+    val symbol = getSymbolFromLine(line)
+    definitionProvider.fromSymbol(convert(symbol)).asScala.headOption.map {
```

Comment:
Why won't we just jump to the line? What is the definition location needed for if we adjust it anyway?

Reply:
to find out what file are we talking about

Reply:
But we do have that information in the stack trace, no?

Reply:
ach, ok I get it now. It's just the name of the file and not the full path, that's why you need it here.